### PR TITLE
Rearrange crates

### PR DESCRIPTION
**Pull request by farmaazon**
### Pull Request Description
This PR is a part of Refactoring Crates initiative. The crates in our repo were rearranged to better reflect the structure described in #448.

The graph_editor and debug_scenes were merged to one "IDE view" crate.

### Important Notes
This PR don't change or move any code - that is a part of #597 and #598 , so for example the debug scenes are not split between IDE and EnsoGL, also the controller crate is currently empty.

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Rust](https://github.com/luna/enso/blob/main/docs/style-guide/rust.md).
- [x] All code has been tested where possible.
- [x] All code has been profiled where possible.

### CODE REVIEW

File: src/rust/ensogl/lib/msdf-sys/Cargo.toml [hunk 1]
```diff
@@ -8,15 +8,15 @@ edition = "2018"
...
 
 [dev-dependencies]
 wasm-bindgen-test          = { version = "0.3.8"                           }
 futures                    = { version = "0.3.1"                           }
 ensogl-core-embedded-fonts = { version = "0.1.0", path="../embedded-fonts" }
 
 [build-dependencies]
-ensogl-build-utilities = { version = "0.1.0", path="../../build" }
\ No newline at end of file
+ensogl-build-utilities = { version = "0.1.0", path="../../../build" }
```
Comment (mwu-tow) [inline src/rust/ensogl/lib/msdf-sys/Cargo.toml:22]: missing newline

File: src/rust/ide/Cargo.toml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [package]
-name    = "ide"
+name    = "enso-ide"
```
Comment (mwu-tow) [inline src/rust/ide/Cargo.toml:2]: We don't have any other IDEs on this repo that we need to differentiate from and I prefer simpler names.
What's the rationale for this rename?

File: src/rust/ide/lib/view/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+wasm-bindgen = { version = "=0.2.58" , features = ["nightly","serde-serialize"] }
+nalgebra     = { version = "0.21.1"  , features = ["serde-serialize"]           }
+serde_json   = { version = "1.0"                                                }
+serde        = { version = "1.0"     , features = ["derive"]                    }
+js-sys       = { version = "0.3.28"                                             }
+
+[dependencies.web-sys]
+version = "0.3.4"
+features = []
```
Comment (mwu-tow) [inline src/rust/ide/lib/view/Cargo.toml:27]: missing newline

File: src/rust/ide/lib/controller/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,7 @@
+#[cfg(test)]
+mod tests {
```
Comment (mwu-tow) [inline src/rust/ide/lib/controller/src/lib.rs:2]: Do we want to version this stub?

File: docs/repository-structure.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+* `build` - Helper JS scripts used by `run` script in the main directory.  
+* `docs` - Documentation.
+* `src/js` - the JS part of IDE application.
+* `src/rust` - All Rust crates.
+
+## Crates
+
+The crates structure is currently in process of refactoring, and there are still places where
+code does not fit the crate it is placed. The work is tracked in #448 epic.
```
Comment (mwu-tow) [inline docs/repository-structure.md:11]: It should be made clear if this document is about the "current" state or about the "desired" state.
Also, I'd be nice to shortly describe the difference between the two.

Comment (farmaazon) [reply docs/repository-structure.md:11]: But the work is very dynamic. I rather assumed, that the user will see an epic summary and the progress of tasks of those epic.

It's quite obvious, that non-closed epic describes the _desired_ state, isn't it?

**Comment by farmaazon** [comment]
@mwu-tow @wdanilo please review it, I'm about to merge it.

File: src/rust/ide/lib/view/src/debug_scenes/shapes.rs [hunk 6]
```diff
@@ -163,10 +163,12 @@ fn init(app:&Application) {
 // === Mocks ===
 // =============
 
+use crate::graph_editor::component::node::port::Expression;
+
 use ast::crumbs::PatternMatchCrumb::*;
 use ast::crumbs::*;
 use span_tree::traits::*;
-use graph_editor::component::node::port::Expression;
+use ensogl_core_msdf_sys::run_once_initialized;
```
Comment (mwu-tow) [inline src/rust/ide/lib/view/src/debug_scenes/shapes.rs:171]: use decls order

File: src/rust/ide/lib/view/src/debug_scenes/visualization.rs [hunk 7]
```diff
@@ -1,13 +1,14 @@
 //! This is a visualization example scene which creates a sinusoidal graph.
 
+use crate::graph_editor::data;
```
Comment (mwu-tow) [inline src/rust/ide/lib/view/src/debug_scenes/visualization.rs:3]: order

File: src/rust/ide/lib/view/src/graph_editor.rs [hunk 8]
```diff
@@ -154,7 +134,7 @@ where T:Eq+Hash, S:Default+std::hash::BuildHasher {
 }
 
 impl<T,S> SharedHashSet<T,S>
-where T:Eq+Hash, S:std::hash::BuildHasher {
+    where T:Eq+Hash, S:std::hash::BuildHasher {
```
Comment (mwu-tow) [inline src/rust/ide/lib/view/src/graph_editor.rs:137]: here and below — indentation

File: src/rust/ide/lib/view/src/graph_editor.rs [hunk 9]
```diff
@@ -445,15 +425,15 @@ impl FrpInputs {
...
-             ,remove_all_node_input_edges,remove_all_node_output_edges,set_visualization_data
-             ,set_detached_edge_targets,set_edge_source,set_edge_target
-             ,unset_edge_source,unset_edge_target
-             ,set_node_position,set_expression_type,select_node,remove_node,set_node_expression
-             ,connect_nodes,deselect_all_nodes,cycle_visualization,set_visualization
-             ,register_visualization,some_edge_targets_detached,some_edge_sources_detached
-             ,all_edge_targets_attached,hover_node_input,all_edge_sources_attached
-             ,hover_node_output,press_node_output,set_detached_edge_sources,all_edges_attached
-             }
+            ,remove_all_node_input_edges,remove_all_node_output_edges,set_visualization_data
```
Comment (mwu-tow) [inline src/rust/ide/lib/view/src/graph_editor.rs:428]: undesired indentation change

File: src/rust/ide/lib/view/src/graph_editor.rs [hunk 10]
```diff
@@ -922,9 +902,9 @@ impl GraphEditorModelWithNetwork {
 
     fn new_node
     ( &self
-    , cursor_style : &frp::Source<cursor::Style>
-    , output_press : &frp::Source<NodeId>
-    , input_press  : &frp::Source<EdgeTarget>
+      , cursor_style : &frp::Source<cursor::Style>
```
Comment (mwu-tow) [inline src/rust/ide/lib/view/src/graph_editor.rs:905]: undesired indentation change

File: src/rust/ide/lib/view/src/graph_editor.rs [hunk 11]
```diff
@@ -1406,7 +1386,7 @@ impl application::View for GraphEditor {
 
 fn enable_disable_toggle
 (network:&frp::Network, enable:&frp::Source, disable:&frp::Source, toggle:&frp::Source)
--> frp::Stream<bool> {
+ -> frp::Stream<bool> {
```
Comment (mwu-tow) [inline src/rust/ide/lib/view/src/graph_editor.rs:1389]: undesired indentation change

File: src/rust/ide/lib/view/src/graph_editor/builtin/visualization/native/bubble_chart.rs [hunk 12]
```diff
@@ -2,10 +2,10 @@
 
 use crate::prelude::*;
 
-use crate::component::visualization::*;
-use crate::component::visualization;
-use crate::frp;
+use crate::graph_editor::component::visualization::*;
+use crate::graph_editor::component::visualization;
```
Comment (mwu-tow) [inline src/rust/ide/lib/view/src/graph_editor/builtin/visualization/native/bubble_chart.rs:6]: not new but somewhat puzzling — why do we import both module and everything from the module?

File: src/rust/ide/src/lib.rs [hunk 13]
```diff
@@ -70,3 +73,21 @@ pub mod prelude {
     #[cfg(test)] pub use wasm_bindgen_test::wasm_bindgen_test;
     #[cfg(test)] pub use wasm_bindgen_test::wasm_bindgen_test_configure;
 }
+
+/// IDE startup function.
+#[wasm_bindgen]
+#[allow(dead_code)]
+pub fn run_example_ide() {
```
Comment (mwu-tow) [inline src/rust/ide/src/lib.rs:80]: this sounds as if we were running "example ide" in opposition to some "real ide",
But then this is not what the docstring says

Comment (farmaazon) [reply src/rust/ide/src/lib.rs:80]: That is a bigger problem; the thing is that our current launcher in js is looking for all functions wilth name "run_example_<something>". And we pick "ide' by default.

This should be changed, and I'll create a task for it.

**Pull request merged**